devito/passes/iet/definitions.py

```diff
@@ -556,3 +579,18 @@ def process(self, graph):
         self.place_devptr(graph)
         self.place_bundling(graph)
         self.place_casts(graph)
+
+
+def make_zero_init(obj):
+    cdims = []
+    for d, (h0, h1), s in zip(obj.dimensions, obj._size_halo, obj.symbolic_shape):
```

Comment:
I thought we already had a version of that in `initialize_function` no?

Reply:
There is some overlap with `initialize_function`, although in that case the initialisation is applied in-place when the function is called using the host-side buffer. In the case here, the initialisation is called within the operator. Not sure if there is some code redundancy that could be avoided?

Reply:
yeah, this **might** be one of those rare cases in which it's preferable to keep them separated, as there are minor differences which would make generalisation probably more tricky to understand and maintain than anything... but open for debate